src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/GeneralPreferencesPane.java

```diff
@@ -591,16 +591,17 @@ public RestartRequirement onApply(UserPrefs prefs)
       if (BrowseCap.isElectron())
       {
          boolean useNativeDialogsPrefValue = nativeFileDialogs_.getValue();
+         boolean useWebDialogsPrefValue = WebDialogCookie.getUseWebDialogs();
          if (useNativeDialogsPrefValue != initialNativeFileDialogs_)
          {
             restartRequirement.setUiReloadRequired(true);
          }
 
-         // The uiLanguage preference is mirrored in a cookie telling GWT which language to display.
+         // The choice of native versus web dialogs is mirrored in a cookie telling GWT which dialogs to use
          // Ensure consistency here and force a page reload if necessary.
-         if (WebDialogCookie.getUseWebDialogs() != useNativeDialogsPrefValue)
+         if (useWebDialogsPrefValue == useNativeDialogsPrefValue)
```

Comment:
`useWebDialogsPrefValue == useNativeDialogsPrefValue` is unintuitive. Maybe it would be clearer if `useNativeDialogsPrefValue` was renamed.

`boolean useWebDialogsPrefValue = !nativeFileDialogs_.getValue();`

Then comparing the desired value to the cookie wouldn't be so confusing.